recipes/omr1/pipeline/tir_solution_generation.py

```diff
@@ -0,0 +1,277 @@
+    run_after = f"{expname}-fill-majority-{suffix}"
+
+    generate(
+        ctx=wrap_arguments(f"++input_dir={output_dir}/filled-majority-{suffix} {extra_args} "),
+        cluster=cluster,
+        generation_type="math_judge",
+        output_dir=f"{output_dir}/judged-generations-{suffix}",
+        expname=f"{expname}-judge-answers-{suffix}",
+        run_after=run_after,
+        **generate_kwargs,
+    )
+
+
+def preprocess_tir_generations(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    cmd = (
+        f"python /nemo_run/code/recipes/omr1/scripts/preprocess_tir_generations.py "
+        f"--input_files '{output_dir}/judged-generations-{suffix}/output-rs*.jsonl' "
+        f"--output_file {output_dir}/tir-preprocess-generations-{suffix}/preprocessed_output.jsonl "
+        f"--code_begin '```python' "
+        f"--code_end '```' "
+    )
+    run_cmd(
+        ctx=wrap_arguments(cmd),
+        cluster=cluster,
+        partition="cpu",
+        log_dir=f"{output_dir}/tir-preprocess-generations-{suffix}/logs",
+        exclusive=True,
+        run_after=f"{expname}-judge-answers-{suffix}",
+        expname=f"{expname}-tir-preprocess-generations-{suffix}",
+    )
+
+
+def filter_novelty_significance(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    run_after = f"{expname}-fill-majority-{suffix}"
+    fragments_file = f"{output_dir}/tir-filter-novelty-significance-{suffix}/output_fragments.jsonl"
+    novelty_output_dir = f"{output_dir}/tir-filter-novelty-significance-{suffix}/novelty_judges/"
+    significance_output_dir = f"{output_dir}/tir-filter-novelty-significance-{suffix}/significance_judges/"
+    dependecies = []
+
+
+    extraction_cmd = (
+        f"python /nemo_run/code/recipes/omr1/scripts/extract_python_fragments.py "
+        f"--input_file={output_dir}/tir-preprocess-generations-{suffix}/preprocessed_output.jsonl "
+        f"--output_file={fragments_file} "
+        f"--window_size=1500"
+    )
+    
+    expname = f"{expname}-tir-extract-python-fragments-{suffix}"
+    run_cmd(
+        ctx=wrap_arguments(extraction_cmd),
+        cluster=cluster,
+        partition="cpu",
+        run_after=run_after,
+        log_dir=f"{output_dir}/tir-filter-novelty-significance-{suffix}/logs",
+        exclusive=True,
+        expname=expname,
+    )
+    run_after = expname
+
+    expname = f"{expname}-tir-judge-novelty-{suffix}"
+    generate(
+        ctx=wrap_arguments(
+            f"++prompt_template=qwen-instruct "
+            f"++prompt_config=/nemo_run/code/recipes/omr1/prompts/classify-tir-novelty.yaml "
+            f"++batch_size=512 "
+            f"++input_file={fragments_file} "
+            f"++generation_key=fragment_novelty "
+            f"++skip_filled=True "
+        ),
+        cluster=cluster,
+        output_dir=novelty_output_dir,
+        run_after=run_after,
+        model="/trt_models/qwen2.5-32b-instruct",
```

Comment:
best to put all that in config like in other pipelines

Reply:
the input / output dirs can be left, but the model / num gpus / num chunks should ideally be easy to change from the config